keps/sig-cloud-provider/0013-build-deploy-ccm.md

```diff
@@ -0,0 +1,496 @@
+cloud providers will place the relevant code in these repos. Whether this is a long-term solution to which additional 
+cloud providers will be added, or an incremental step toward moving out of the Kubernetes org is out of scope of this 
+document, and merits discussion in a broader forum and input from SIG-Architecture and Steering Committee. 
+- **K8s SIGs/library**: Any SIG owned repository. 
+- **Staging**: Staging: Separate repositories which are currently visible under the K8s/K8s repo, which contain code
+considered to be safe to be vendored outside of the K8s/K8s repo and which should eventually be fully separated from 
+the K8s/K8s repo. Contents of Staging are prevented from depending on code in K8s/K8s which are not in Staging. 
+Controlled by [publishing kubernetes-rules-configmap](https://github.com/kubernetes/publishing-bot/blob/master/configs/kubernetes-rules-configmap.yaml)
+
+
+## Summary
+
+Cloud provider specific logic will be removed from the K8s/K8s repository and the generic cloud provider code will be restructured 
+to make it easy for any cloud provider to extend the kubernetes core in a consistent manner for their cloud. New cloud
+providers should look at the [Creating a Custom Cluster from Scratch](https://kubernetes.io/docs/getting-started-guides/scratch/#cloud-provider)
+and the [cloud provider interface](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/cloud.go#L31)
+which will need to be implemented.
+
+## Motivation
+
+As the Kubernetes ecosystem grows and more cloud providers implement the cloud provider interface, the number of
+dependencies in the Kubernetes core code will increase too, increasing bloat and complexity. One improvement that
+can be made to decrease these dependencies is removing the cloud provider implementations from core and into their
+own cloud provider owned repositories.
+
+Currently there are eight cloud provider implementations. To prevent this number from growing we have locked Kubernetes
+Core to the addition of any new dependencies. This means all new cloud providers have to implement all their pieces
+outside of the Core. However, the current cloud providers are still used by core Kubernetes components. In order to
+remove the dependencies, changes are required in the following areas:
+
+- [Kube Controller Manager](https://kubernetes.io/docs/reference/generated/kube-controller-manager/) - Track usages of [CMServer.CloudProvider](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/options/options.go)
+- [API Server](https://kubernetes.io/docs/reference/generated/kube-apiserver/) - Track usages of [ServerRunOptions.CloudProvider](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go)
+- [Kubelet](https://kubernetes.io/docs/reference/generated/kubelet/) - Track usages of [KubeletFlags.CloudProvider](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go)
```

Comment:
`Kubelet` -> `kubelet` in the doc.
except if it's about API types such as `KubeletFlags`

Reply:
Fixed